### DISCUSSION TITLE
SharpGen.Runtime.SharpGenException:“HRESULT: [0x80070057], Module: [General], ApiCode: [E_INVALIDARG/Invalid arguments], Message: [参数错误。 ]”

### DISCUSSION THREAD

**Discussion by Hommee-45** [amerkoleci/Vortice.Windows#442] [General]
` 
 private void BuildRootSignature()
        {

            DescriptorRange1 texTableRange = new DescriptorRange1(DescriptorRangeType.ShaderResourceView, 4, 0);
            RootDescriptorTable1 texTable = new RootDescriptorTable1(texTableRange);

            DescriptorRange1 texTableRange1 = new DescriptorRange1(DescriptorRangeType.ShaderResourceView, 5, 1);
            RootDescriptorTable1 texTable1 = new RootDescriptorTable1(texTableRange1);

            RootDescriptor1 rootDescriptor1 = new RootDescriptor1(0, 0);
            RootDescriptor1 rootDescriptor2 = new RootDescriptor1(1, 0);
            RootDescriptor1 rootDescriptor3 = new RootDescriptor1(0, 1);

            RootParameter1[] slotRootParameters = new[]
            {
                new RootParameter1(RootParameterType.ConstantBufferView, rootDescriptor1, ShaderVisibility.All),
                new RootParameter1(RootParameterType.ConstantBufferView, rootDescriptor2, ShaderVisibility.All),
                new RootParameter1(RootParameterType.ShaderResourceView, rootDescriptor3, ShaderVisibility.All),
                new RootParameter1(texTable, ShaderVisibility.All),
                new RootParameter1(texTable1, ShaderVisibility.All),
            };

            // A root signature is an array of root parameters.
            RootSignatureDescription1 rootSigDesc = new RootSignatureDescription1(
                RootSignatureFlags.AllowInputAssemblerInputLayout,
                slotRootParameters,
                GetStaticSamplers()
                );

            _rootSignature = Device.CreateRootSignature(rootSigDesc);
        }`

final code CreateRootSignature would throw a error 

SharpGen.Runtime.SharpGenException:“HRESULT: [0x80070057], Module: [General], ApiCode: [E_INVALIDARG/Invalid arguments], Message: [参数错误。
]”
![Error1-2](https://github.com/amerkoleci/Vortice.Windows/assets/58999256/7488499d-dc2a-4b9e-ab8c-a45d680d3eb5)
![Error1-1](https://github.com/amerkoleci/Vortice.Windows/assets/58999256/12f027b2-2e57-4b7a-97cf-9c67261e022b)

**Comment by amerkoleci** [comment]
Enable d3d12 debug layer and native debugging so you can see whats wrong on native d3d12 side

**Reply by Hommee-45** [reply]
![Error1-3](https://github.com/amerkoleci/Vortice.Windows/assets/58999256/be15f3d1-b0a7-444b-899a-d862290fdebf)
I disable the second RootDescriptorTable line code,  it would not to cause any problem. Add one more RootDescriptor don't cause any problem too. But add one more  RootDescriptorTable it would throw a exception.

**Comment by Hommee-45** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Hommee-45** [comment]
> Enable d3d12 debug layer and native debugging so you can see whats wrong on native d3d12 side

Thanks for your replying!!How to enable d3d12 debug layer and native, do you have a link?

**Comment by amerkoleci** [comment]
Follow the sample here:
https://github.com/amerkoleci/Vortice.Windows/blob/main/src/samples/HelloDirect3D12/Program.cs

**Reply by Hommee-45** [reply]
![EnableDebug](https://github.com/amerkoleci/Vortice.Windows/assets/58999256/2075f58e-6c92-42fa-93de-d279d0c6e85c)
i debug the line with the opening debug layer code, it does hit the break point. And then what i have to do to debug the native code.

**Comment by MarcioAB** [comment]
Enable native code debugging ...
![image](https://github.com/amerkoleci/Vortice.Windows/assets/15105477/8e4f9ad1-e0a9-4b3e-9e16-d203a06502bf)

**Reply by Hommee-45** [reply]
i have tried it.  I use SharpDX,  it wounldn't cause any problems

**Reply by lindexi** [reply]
@MarcioAB Thank you very much. After enable native code debugging and `DeviceCreationFlags.Debug | DeviceCreationFlags.Debuggable`, I can find the debug message in Visual Studio output window.
